samples/batch/transcription-enabled-storage/Connector/BatchClient.cs

```diff
@@ -0,0 +1,195 @@
+        {
+            SubscriptionKey = subscriptionKey;
+            HostName = hostName;
+            Log = log;
+        }
+
+        public static bool IsThrottledOrTimeoutStatusCode(HttpStatusCode statusCode)
+        {
+            if (statusCode == HttpStatusCode.TooManyRequests ||
+                statusCode == HttpStatusCode.GatewayTimeout ||
+                statusCode == HttpStatusCode.RequestTimeout ||
+                statusCode == HttpStatusCode.BadGateway ||
+                statusCode == HttpStatusCode.NotFound)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public Task<IEnumerable<Transcription>> GetTranscriptionsAsync()
+        {
+            var path = $"{SpeechToTextBasePath}Transcriptions";
+            return this.GetAsync<IEnumerable<Transcription>>(path);
+        }
+
+        public Task<Transcription> GetTranscriptionAsync(Guid id)
+        {
+            var path = $"{SpeechToTextBasePath}Transcriptions/{id}";
+            return this.GetAsync<Transcription>(path);
+        }
+
+        public Task<TranscriptionFiles> GetTranscriptionFilesAsync(Guid id)
+        {
+            var path = $"{SpeechToTextBasePath}Transcriptions/{id}/files";
+            return this.GetAsync<TranscriptionFiles>(path);
+        }
+
+        public Task<Uri> PostTranscriptionAsync(string name, string description, string locale, Dictionary<string, string> properties, List<Uri> contentUrls, IEnumerable<Guid> modelIds)
+        {
+            var models = modelIds.Select(m => ModelIdentity.Create(m)).ToList();
+            var path = $"{SpeechToTextBasePath}Transcriptions/";
+
+            var transcriptionDefinition = TranscriptionDefinition.Create(name, description, locale, contentUrls, properties, models);
+            return this.PostAsJsonAsync(path, transcriptionDefinition);
+        }
+
+        public Task DeleteTranscriptionWithRetryAsync(Guid id)
+        {
+            var path = $"{SpeechToTextBasePath}Transcriptions/{id}";
+            return this.DeleteAsync(path);
+        }
+
+        private static Uri GetLocationFromPostResponseAsync(WebHeaderCollection headers)
+        {
+            return new Uri(headers["Location"]);
+        }
+
+        private static async Task<HttpWebResponse> GetHttpWebResponseAsync(HttpWebRequest request, ILogger logger)
+        {
+            var webresponse = await request.GetResponseAsync().ConfigureAwait(false);
+            var response = (HttpWebResponse)webresponse;
+
+            if (IsThrottledOrTimeoutStatusCode(response.StatusCode))
+            {
+                throw new TimeoutException();
+            }
+
+            if (response.StatusCode != HttpStatusCode.Accepted && response.StatusCode != HttpStatusCode.Created)
+            {
+                logger.LogInformation($"Failure: Status Code {response.StatusCode}, {response.StatusDescription}");
+                throw new NotImplementedException();
```

Comment:
Why throw NotImplementedException?

Reply:
because I didn't add any specific handling for that scenario. But you're right, I've changed it to a web exception and will just catch it in the main function